src/components/transactions/modals/ClaimWRBTCModal.vue

```diff
@@ -159,12 +169,26 @@ export default {
           break
         }
         case this.claimTypes.CONVERT_TO_RBTC: {
-          const weiAmount = new BigNumber(this.receiveAmount).shiftedBy(-8).toString()
+          this.processing = true
+          
+          const weiAmount = new BigNumber(this.receiveAmount).shiftedBy(8).toFixed(0)
+
           this.responseEstimatedGas = await this.getEstimatedGasPrice(weiAmount)
-          const estimatedGas = new BigNumber(this.responseEstimatedGas?.amountEstimatedGas)
+          // TODO: check if response is not null otherwise show an error
+          
+          const estimatedGas = new BigNumber(this.responseEstimatedGas?.amount)
             .shiftedBy(-8)
             .toString()
-          this.receiveAmount = new BigNumber(this.amount).minus(estimatedGas).toString()
+
+          this.sharedState.web3.eth.getGasPrice().then(gasPrice => {
+            const costInWei = new BigNumber(estimatedGas)
+              .multipliedBy(gasPrice)
+              .shiftedBy(-10)
+              .toPrecision(8).toString()
```

Comment:
Usually to convert to wei i use shift by the decimal amounts and then toFixed(0), as precision comes with decimals and amounts in wei can't have decimals